src/pages/spawn/spawnHost/SpawnHostTable.tsx

```diff
@@ -1,3 +1,149 @@
 import React from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import Button, { Size } from "@leafygreen-ui/button";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Table } from "antd";
+import { formatDistanceToNow } from "date-fns";
+import { HostStatusBadge } from "components/HostStatusBadge";
+import Icon from "components/icons/Icon";
+import { Host } from "gql/generated/types";
+import {
+  copyToClipboard,
+  sortFunctionDate,
+  sortFunctionString,
+} from "utils/string";
+import { SpawnHostActionButton } from "./SpawnHostActionButton";
 
-export const SpawnHostTable = () => <>Spawn Host table</>;
+interface SpawnHostTableProps {
+  hosts: Host[];
+}
+export const SpawnHostTable: React.FC<SpawnHostTableProps> = ({ hosts }) => {
+  const columns = [
+    {
+      title: "Host",
+      dataIndex: "id",
+      key: "host",
+      sorter: (a, b) => sortFunctionString(a, b, "id"),
+      render: (_, host: Host) => <HostIdField host={host} />,
+    },
+    {
+      title: "Distro",
+      dataIndex: "distro",
+      key: "distro",
+      sorter: (a, b) => sortFunctionString(a, b, "distro.id"),
+      render: (distro) => distro.id,
+    },
+    {
+      title: "Status",
+      dataIndex: "status",
+      key: "status",
+      sorter: (a, b) => sortFunctionString(a, b, "status"),
+      render: (status) => <HostStatusBadge status={status} />,
+    },
+    {
+      title: "Expires In",
+      dataIndex: "expiration",
+      key: "expiration",
+      sorter: (a, b) => sortFunctionDate(a, b, "expiration"),
+      render: (expiration) =>
+        formatDistanceToNow(new Date(expiration)) || "Does not expire",
```

Comment:
I almost left the opposite comment on Chaya's pr, but this function is less precise than what we had before, and I think is not quite precise enough for these data elements. Could you use the function that she created to format durations? Probably need to coordinate some git logistics depending on who merges first

Reply:
Just left a comment on her pr https://github.com/evergreen-ci/spruce/pull/343#discussion_r471778881 depending on when it is merged i can add this in a follow up pr.

Reply:
I don't think it's a good idea to modify stringifyNanoseconds to fit this use case. stringifyNanoseconds takes a number as input, for examples something like 1183, and returns something like < 1 ms. This use case takes a date object pointing at some time in the future, and tries to determine how much time is left between now and then.

Reply:
I agree stringifyNanoseconds isnt suited for this use case. I think using formatDistanceToNow is is fine if we end up needing more granularity in the future we can probably create our own function but i dont think its as urgent.